master/internal/experiment.go

```diff
@@ -124,6 +124,7 @@ func newExperiment(
 	expModel *model.Experiment,
 	activeConfig expconf.ExperimentConfig,
 	taskSpec *tasks.TaskSpec,
+	restored bool,
```

Comment:
I think this line here is checking if the experiment is not restored

I think it makes sense to avoid passing in restore and using this ```exp.model.id``` for this check or make them both check the boolean 

https://github.com/determined-ai/determined/pull/7113/files#diff-d92b0050d5549a1efe86c524037f8a2709d6ebf8ce2c63a777f63744bc8a0b00R167